src/code/utilities/authoring-utils.js

```diff
@@ -0,0 +1,51 @@
+export default class AuthoringUtils {
+  static getChallengeDefinition(authoring, routeSpec) {
+    const challengeId = this.getChallengeId(authoring, routeSpec),
+          challenges = authoring.challenges;
+    return challengeId && challenges ? authoring.challenges[challengeId] : null;
+  }
+
+  static getChallengeId(authoring, routeSpec) {
+    const levels = authoring.application.levels,
+          missions = levels ? levels[routeSpec.level].missions : null,
+          challenges = missions ? missions[routeSpec.mission].challenges : null,
+          challenge = challenges ? challenges[routeSpec.challenge] : null;
+    return challenge ? challenge.id : null;
```

Comment:
This is a good point- I hadn't actually realized `return challenge && challenge.id;` would return null anyway, so I agree it's the cleaner solution here.